### PR TITLE
TGQuery refactoring

### PR DESCRIPTION
**Pull request by dlsniper**

### CODE REVIEW

**Comment by dlsniper** [comment]
@nilsen340 have a look at this beauty :D

**Comment by nilsen340** [comment]
@dlsniper looking =)

File: app/app.iml [hunk 1]
```diff
@@ -104,8 +109,8 @@
     <orderEntry type="library" exported="" scope="TEST" name="espresso-idling-resource-2.2.1" level="project" />
     <orderEntry type="library" exported="" name="okio-1.6.0" level="project" />
     <orderEntry type="library" exported="" scope="TEST" name="runner-0.4.1" level="project" />
-    <orderEntry type="library" exported="" name="gson-2.5" level="project" />
     <orderEntry type="library" exported="" name="logging-interceptor-2.7.0" level="project" />
+    <orderEntry type="library" exported="" name="gson-2.5" level="project" />
     <orderEntry type="library" exported="" name="retrofit-2.0.0-beta2" level="project" />
     <orderEntry type="library" exported="" scope="TEST" name="javax.annotation-api-1.2" level="project" />
     <orderEntry type="library" exported="" scope="TEST" name="hamcrest-library-1.3" level="project" />
```
Comment (nilsen340) [inline app/app.iml:116]: probably we should add this file to gitignore

File: tapglue-android-sdk/tapglue-android-sdk.iml [hunk 2]
```diff
@@ -96,12 +96,12 @@
...
     <orderEntry type="library" exported="" scope="TEST" name="hamcrest-core-1.3" level="project" />
+    <orderEntry type="library" exported="" name="logging-interceptor-2.7.0" level="project" />
     <orderEntry type="library" exported="" name="retrofit-2.0.0-beta2" level="project" />
     <orderEntry type="library" exported="" scope="TEST" name="junit-4.12" level="project" />
-    <orderEntry type="library" exported="" name="okhttp-2.7.0" level="project" />
     <orderEntry type="library" exported="" name="support-annotations-23.1.1" level="project" />
+    <orderEntry type="library" exported="" name="okhttp-2.7.0" level="project" />
     <orderEntry type="library" exported="" name="converter-gson-2.0.0-beta2" level="project" />
   </component>
 </module>
```
Comment (nilsen340) [inline tapglue-android-sdk/tapglue-android-sdk.iml:108]: this one probably also for gitignore

Comment (dlsniper) [inline tapglue-android-sdk/tapglue-android-sdk.iml:108]: Yeah, will do a cleanup PR for this separately.

File: app/src/main/java/com/tapglue/tapgluetests/TestActivity.java [hunk 3]
```diff
@@ -600,8 +603,26 @@ public void onRequestFinished(@Nullable TGFeed output, boolean changeDoneOnline)
                         }
 
                         testController.log("#3.5 finished correctly");
-                        // TODO: after fix of delete request change this to TEST_3_6
-                        doTest(TEST_3_7_1, randomUserName, randomUserName2, runnable);
+                        TGQuery query = new TGQuery();
+                        query.setObject(new TGQueryTGObject().setId(new TGQueryObject<String>().setEq("hello")));
```
Comment (nilsen340) [inline app/src/main/java/com/tapglue/tapgluetests/TestActivity.java]: Im a little confused, whats the difference between TGQueryObject and TGQueryTGObject ??

**Comment by dlsniper** [comment]
Right now the query looks like this:

``` java
import static com.tapglue.model.queries.TGQuery.*;

ArrayList<String> inValues = new ArrayList<>();
inValues.add("hello");
inValues.add("world");

TGQuery query = new TGQuery();
query.addConstraint(type(eq("hello")));
query.addConstraint(tgObjectId(in(inValues)));
query.addConstraint(object(objectId(eq("hello"))));
query.addConstraint(object(objectType(in("hello", "world"))));
```

with this output:

``` json
{"object":{"id":{"in":["hello","world"]}},"tg_object_id":{"in":["hello","world"]},"type":{"eq":"hello"}}
```

Which has a bug but it's easy to handle / fix.

**Pull request merged**